TranslatorReasonerAPI.yaml

```diff
@@ -682,3 +700,51 @@ components:
         - predicate
         - object
       additionalProperties: false
+    QueryConstraint:
+      type: object
+      description: >-
+        Generic query constraint for a query node or query edge
+      properties:
+        name:
+          type: string
+          description: >-
+            Human-readable name or label for the constraint. If appropriate,
+            it should be the name of the semantic type term.
+          example: molecular mass
+          nullable: false
+        id:
+          allOf:
+            - $ref: '#/components/schemas/CURIE'
+          description: >-
+            CURIE of the concept being constrained. For properties
+            defined by the Biolink model this should be a biolink CURIE,
+            otherwise, if possible, from the EDAM ontology. If a suitable
+            identifier does not exist, enter a descriptive phrase here and
+            submit the new type for consideration by the appropriate
+            authority.
+          example: EDAM:data_0844
+          nullable: false
+        not:
+          type: boolean
+          default: false
+        operator:
+          type: string
+          enum:
+            - ==
+            - '>'
+            - <
+            - in
+            - matches
```

Comment:
I think that this probably requires a description field to explain the various conditions and how they might behave given different `value` types.